api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1153,8 +1168,166 @@ public Appointment bookAppointment(Appointment appointment,
+						.getAppointmentType().getDuration()) {
+					throw new TimeSlotFullException();
+				}
+			}
+
+			appointment.setStatus(AppointmentStatus.SCHEDULED);
+			return Context.getService(AppointmentService.class).saveAppointment(
+					appointment);
+		}
+
+	}
+
+
+	//appointment resource
+
+	@Override
+	public List<AppointmentResource> getAllAppointmentResources() {
+		return getAppointmentResourceDAO().getAll();
+	}
+
+	@Override
+	public List<AppointmentResource> getAllAppointmentResources(boolean includeVoided) {
+		return getAppointmentResourceDAO().getAllData(includeVoided);
+	}
+
+	@Override
+	public AppointmentResource getAppointmentResource(Integer appointmentResourceId) {
+		return (AppointmentResource) getAppointmentResourceDAO().getById(appointmentResourceId);
+	}
+
+	@Override
+	public AppointmentResource getAppointmentResourceByUuid(String uuid) {
+		return (AppointmentResource) getAppointmentResourceDAO().getByUuid(uuid);
+	}
+
+	@Override
+	public AppointmentResource saveAppointmentResource(AppointmentResource appointmentResource) throws APIException {
+		appointmentResource.setIncludeWeekends(true);
+		ValidateUtil.validate(appointmentResource);
+		return saveAppointResourceInternal(appointmentResource);
+	}
+
+	@Override
+	public AppointmentResource voidAppointmentResource(AppointmentResource appointmentResource, String reason) {
+		appointmentResource.setVoided(true);
+		appointmentResource.setVoidReason(reason);
+		return (AppointmentResource) getAppointmentResourceDAO().saveOrUpdate(appointmentResource);
 	}
 
+	@Override
+	public void purgeAppointmentResource(AppointmentResource appointmentResource) {
+		getAppointmentResourceDAO().delete(appointmentResource);
+	}
+
+	@Override
+	public List<AppointmentResource> getAppointmentResourcesByConstraints(Location location, Provider provider, List<AppointmentType> appointmentTypes) {
+		return getAppointmentResourceDAO().getResourceByConstraints(null, null, null, null, location, null);
+	}
+
+	@Override
+	public BlockExcludedDays geyExcludedDayByUuid(String uuid) throws APIException {
+		return appointmentResourceDAO.geyExcludedDayByUuid(uuid);
+	}
+
+	private synchronized AppointmentResource saveAppointResourceInternal(AppointmentResource appointmentResource) throws APIException {
+		//check if resource already exists
+		List<AppointmentResource> resource = getAppointmentResourceDAO().getResourceByConstraints(appointmentResource.getStartDate(), appointmentResource.getEndDate(),
+				appointmentResource.getStartTime(), appointmentResource.getEndTime(), appointmentResource.getLocation(), null);
+		if (resource.size() > 0) {
+			throw new APIException("Resource with those details already exist");
+		} else {
+			if (CollectionUtils.isEmpty(appointmentResource.getDaysList())) {
+				return (AppointmentResource) getAppointmentResourceDAO().saveOrUpdate(appointmentResource);
+			}
+			appointmentResource.setDaysList(null);
```

Comment:
We are always setting the days list to null upon saving?  This seems confusing, or am I missing something?